src/bin/directory-cli.rs

```diff
@@ -22,30 +26,28 @@ enum Commands {
     ListAddresses,
 }
 
-fn send_rpc_req(req: &RpcMsgReq) {
-    let mut stream = TcpStream::connect("127.0.0.1:4321").unwrap();
-    stream
-        .set_read_timeout(Some(Duration::from_secs(20)))
-        .unwrap();
-    stream
-        .set_write_timeout(Some(Duration::from_secs(20)))
-        .unwrap();
+fn send_rpc_req(req: &RpcMsgReq) -> Result<(), DirectoryServerError> {
+    let mut stream = TcpStream::connect("127.0.0.1:4321")?;
+    stream.set_read_timeout(Some(Duration::from_secs(20)))?;
+    stream.set_write_timeout(Some(Duration::from_secs(20)))?;
 
-    send_message(&mut stream, &req).unwrap();
+    send_message(&mut stream, &req)?;
 
-    let resp_bytes = read_message(&mut stream).unwrap();
-    let resp: RpcMsgResp = serde_cbor::from_slice(&resp_bytes).unwrap();
+    let resp_bytes = read_message(&mut stream)?;
+    let resp: RpcMsgResp = serde_cbor::from_slice(&resp_bytes).map_err(NetError::Cbor)?;
 
     println!("{:?}", resp);
+    Ok(())
 }
 
-fn main() {
+fn main() -> Result<(), DirectoryServerError> {
```

Comment:
Yes. But I would prefer a `?` than `unwrap()` in the code. In expense of bigger function definition. It doesn't matter in main. But removes the unwraps. Error messages won't be necessary here and will make the code unnecessarily verbose.

Reply:
That is because the main function there is small and doesn't have many unwraps. It was discussed in BDK dev whether to return the error with `anyhow`. It's in general not always a bad pattern to have return in main. 

In our case, there are too many unwraps. Instead of panicking, outputting the error from the app can make the display for users better. Also reducing code verbosity. Thus it makes more sense here.